src/node_file.cc

```diff
@@ -2748,6 +2748,11 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
   CHECK_GE(argc, 2);
 
   BufferValue tmpl(isolate, args[0]);
+  const char* ex = "XXXXXX";
+  const auto length = tmpl.length();
+  tmpl.AllocateSufficientStorage(length + strlen(ex));
+  snprintf(tmpl.out() + length, length, "%s", ex);
```

Comment:
The second argument, isn't it supposed to be strlen(ex) instead of length?

From https://nodejs.org/api/fs.html#fspromisesmkdtempprefix-options

I understand that it would append 6 random chars, which we have "xxxxxx" as placeholder.

Trying to see a case when prefix is less than 6 characters, it will only copy length count characters.

Sorry If I am wrong :)

Reply:
Yes, you're right! Thank you for following up and leaving a review. I really appreciate it!